src/code/z_actor.c

```diff
@@ -3571,21 +3611,29 @@ void func_800BC154(PlayState* play, ActorContext* actorCtx, Actor* actor, u8 act
 }
 
 // Damage flags for EnArrow
-s32 D_801AEDB0[] = {
-    0x800, 0x20, 0x20, 0x800, 0x1000, 0x2000, 0x1, 0x10000, 0x1,
+u32 sArrowDmgFlags[] = {
+    DMG_FIRE_ARROW,   // ENARROW_0
+    DMG_NORMAL_ARROW, // ENARROW_1
+    DMG_NORMAL_ARROW, // ENARROW_2
+    DMG_FIRE_ARROW,   // ENARROW_3
+    DMG_ICE_ARROW,    // ENARROW_4
+    DMG_LIGHT_ARROW,  // ENARROW_5
+    DMG_DEKU_NUT,     // ENARROW_6
+    DMG_DEKU_BUBBLE,  // ENARROW_7
+    DMG_DEKU_NUT,     // ENARROW_8
 };
 
-s32 func_800BC188(s32 index) {
-    if ((index < 0) || (index >= ARRAY_COUNT(D_801AEDB0))) {
+u32 Actor_GetArrowDmgFlags(s32 params) {
+    if ((params < 0) || (params >= ARRAY_COUNT(sArrowDmgFlags))) {
         return 0;
     }
 
-    return D_801AEDB0[index];
+    return sArrowDmgFlags[params];
 }
 
-s32 func_800BC1B4(Actor* actor, Actor* arg1, f32 arg2, f32 arg3) {
-    if ((arg3 > 0.0f) && (Actor_WorldDistXYZToActor(arg1, actor) < ((arg3 * 2.5f) + arg2))) {
-        s16 temp_v1 = BINANG_SUB(Actor_WorldYawTowardActor(arg1, actor), arg1->world.rot.y);
+s32 func_800BC1B4(Actor* actor, Actor* projectile, f32 distance, f32 speedXZ) {
+    if ((speedXZ > 0.0f) && (Actor_WorldDistXYZToActor(projectile, actor) < ((speedXZ * 2.5f) + distance))) {
```

Comment:
Is this speedXZ or just speed? It is being compared to `Actor_WorldDistXYZToActor` although internally within an expression

Reply:
this will depend on what actor the speed is coming from, if they call `Actor_MoveXZGravity` or `Actor_MoveXYZ`.
From a cursory glance it seems that its generalized enough to allow either, so `speed` would be safer. 

If on closer inspection there is something in this whole projectile system that requires it to use XZGravity in order to work properly, then can you assume.

Reply:
Following the code a little bit seems like I named this after the current `speedXZ` member in `Actor` which will be renamed in #1186 
So I'll rename this parameter to `speed`